src/ui/DynamicHierarchicalFacet/DynamicHierarchicalFacetValues/DynamicHierarchicalFacetValues.ts

```diff
@@ -28,6 +29,16 @@ export class DynamicHierarchicalFacetValues implements IDynamicHierarchicalFacet
     this.facetValues = response.values.map(responseValue => this.createFacetValueFromResponse(responseValue));
   }
 
+  public reorderValues(pathsOrder: string[][]) {
+    const order = pathsOrder.map(path => HashUtils.encodeArray(path));
```

Comment:
`encodeArray` could be replaced with anything else that generates a unique string from a string array.

The reason I'm converting an array to a unique string is to use it as a key in an object, which simplifies sorting. If you dislike this approach, I can change `reorderValuesByKeys` for something else that would also work for paths.